client.go

```diff
@@ -42,48 +86,120 @@ func (c *ExtensionManagerClient) Close() {
 	}
 }
 
-// Ping requests metadata from the extension manager.
+// Ping requests metadata from the extension manager, using a new background context
 func (c *ExtensionManagerClient) Ping() (*osquery.ExtensionStatus, error) {
-	return c.Client.Ping(context.Background())
+	return c.PingContext(context.Background())
 }
 
-// Call requests a call to an extension (or core) registry plugin.
+// PingContext requests metadata from the extension manager.
+func (c *ExtensionManagerClient) PingContext(ctx context.Context) (*osquery.ExtensionStatus, error) {
+	if err := c.lock.Lock(ctx); err != nil {
+		return nil, err
+	}
+	defer c.lock.Unlock()
+	return c.client.Ping(ctx)
+}
+
+// Call requests a call to an extension (or core) registry plugin, using a new background context
 func (c *ExtensionManagerClient) Call(registry, item string, request osquery.ExtensionPluginRequest) (*osquery.ExtensionResponse, error) {
-	return c.Client.Call(context.Background(), registry, item, request)
+	return c.CallContext(context.Background(), registry, item, request)
 }
 
-// Extensions requests the list of active registered extensions.
+// Call requests a call to an extension (or core) registry plugin.
```

Comment:
A couple of the docblocks (here and below) have the old function names instead of the new ones

Reply:
good catch